lib/fluent/plugin/file_wrapper.rb

```diff
@@ -35,20 +35,36 @@ def self.stat(path)
     end
   end
 
-  class WindowsFile
-    require 'windows/file'
-    require 'windows/handle'
+  module Win32API
+    require 'fiddle/import'
+    require 'fiddle/types'
+    extend Fiddle::Importer
+
+    if RUBY_PLATFORM.split('-')[-1] == "ucrt"
+      MSVCRT_DLL = 'ucrtbase.dll'
+    else
+      MSVCRT_DLL = 'msvcrt.dll'
+    end
 
+    dlload MSVCRT_DLL, "kernel32.dll"
+    include Fiddle::Win32Types
+
+    extern "intptr_t _get_osfhandle(int)"
+    extern "BOOL GetFileInformationByHandle(HANDLE, void *)"
+    extern "BOOL GetFileInformationByHandleEx(HANDLE, int, void *, DWORD)"
+  end
```

Comment:
BTW, Can we split this FFI module as a file whose name is win32api.rb?
Scattering FFI code is not good for us.